pkg/runplugin/runplugin.go

```diff
@@ -39,9 +41,27 @@ func (sch *Scheduler) RunPlugin(spec *Spec) error {
 		return fmt.Errorf("invalid plugin name %q", spec.Image)
 	}
 
+	// if no given name for the plugin, use PLUGIN-VERSION-INSTANCE format for name
+	// INSTANCE is calculated as Sha256("DOMAIN/PLUGIN:VERSION&ARGUMENTS") and
+	// take the first 8 hex letters.
+	// NOTE: if multiple plugins with the same version and arguments are given for
+	//       the same domain, only one deployment will be applied to the cluster
+	// NOTE2: To comply with RFC 1123 for Kubernetes object name, only lower alphanumeric
+	//        characters with '-' is allowed
+	name := strings.ToLower(spec.Name)
```

Comment:
I think I'd feel better if instead of silently "fixing" this, we just return an error.

For example, you could do something like:
```go
var validNamePattern = regexp.MustCompile("^[a-z0-9-]+$")

...

func RunPlugin(spec *Spec) {
        ...
	if !validNamePattern.MatchString(spec.Name) {
		return fmt.Errorf("Invalid plugin name %q. (Names must only use lowercase alphanumeric or -)", spec.Name)
	}
        ...
}
```